### ISSUE TITLE
Summary of Google Summer of Code 2021 with pvlib

### ISSUE THREAD

**Issue by AdamRJensen** [pvlib/pvlib-python#1289]
The purpose of this issue is to give a brief summary of the work that I've done during the 2021 [Google Summer of Code (GSoC)](https://summerofcode.withgoogle.com/about/) program with pvlib.

The official project abstract can be found [here](https://summerofcode.withgoogle.com/projects/#6071460558274560), but for a more thorough description, I recommend reading my blog posts which provide a more detailed [project description](https://predictablysunny.com/posts/gsoc_project_intro/) and [evaluation of my experience](https://predictablysunny.com/posts/gsoc_project_summary/). The project was carried out under the mentorship of pvlib maintainer [Kevin Anderson](https://github.com/kanderso-nrel).

In short, the aim of the project was to improve the pvlib iotools by adding file reading and data retrieval capabilities for six additional open-source datasets. Overall it's been a huge success! At the time of writing, the majority of the functions have already been merged into the pvlib master branch.

The code development and discussions are documented in the following pull requests:
* #1175
* #1186
* #1254
* #1264
* #1274
* #1268, see also #1245

Please do not hesitate to contact me regarding articipation in GSoC or the new iotools functions.

**Comment by wholmgren** [comment]
Thanks @AdamRJensen for your great contributions and thanks @kanderso-nrel for mentoring! It's a great set of functionality and I'm sure it will be used by many.
